### PR TITLE
Add classes for Game API. Allow a player to start game.

### PR DESCRIPTION
**Pull request by gordon-cheung**
game

### CODE REVIEW

File: frontend/src/modules/lobby/view/Lobby.vue [hunk 1]
```diff
@@ -45,35 +47,20 @@ export default {
...
+            this.$gameApiService.startGame();
         },
 
         exitToHome: function () {
             this.$webSocketService.disconnect();
             this.$router.push({ name: "home" });
         },
     },
+    created() {
+        EventBus.$on(Event.START_GAME, () => {
```
Comment (bseto) [inline frontend/src/modules/lobby/view/Lobby.vue:60]: ok nice nice i like this

File: frontend/src/services/eventHandlerService.js [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+import EventHandlerFactory from "./eventHandlers/eventHandlerFactory";
 
+// Service for handling websocket events
 export default class EventHandlerService {
     handle(api, payload) {
-        if (api === "hub") {
-            this.handleHubEvent(payload);
+        console.log("----- Event Received -----");
+        console.log(api);
+        console.log(payload);
```
Comment (bseto) [inline frontend/src/services/eventHandlerService.js]: make it so it doesn't log if it's api "draw"

File: frontend/src/services/eventHandlers/hubHandler.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { GlobalStore } from "@/modules/common/store/globalstore";
+
+// Event handler for Hub API
+export default class HubHandler {
```
Comment (bseto) [inline frontend/src/services/eventHandlers/hubHandler.js:4]: what is hub handler?

File: frontend/src/services/eventHandlers/gameHandler.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { EventBus } from "@/eventBus.js";
+import { Event } from "@/events";
+
+// Event handler for Game API
+export default class GameHandler {
```
Comment (bseto) [inline frontend/src/services/eventHandlers/gameHandler.js:5]: is game handler like the game master? or game controller?

File: frontend/src/services/gameApiService.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+// Service for making request to game API.
+export default class GameApiService {
+    constructor(webSocketService) {
+        this.webSocketService = webSocketService;
+    }
+
+    startGame() {
```
Comment (bseto) [inline frontend/src/services/gameApiService.js:7]: nice

File: frontend/src/services/eventHandlerService.js (hunk 2 above)
Comment (gordon-cheung) [reply frontend/src/services/eventHandlerService.js]: move console.log to print only for supported handlers

File: frontend/src/services/eventHandlerService.js [hunk 6]
```diff
@@ -1,18 +1,19 @@
...
+
+        let eventHandlerFactory = new EventHandlerFactory();
```
Comment (sam-ao) [inline frontend/src/services/eventHandlerService.js:7]: I don't think we need to be creating a new instance every time. Maybe the EventHandlerService should maintain a map of handlers

File: frontend/src/services/gameApiService.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+// Service for making request to game API.
```
Comment (sam-ao) [inline frontend/src/services/gameApiService.js:1]: I like this design. I think we should refactor all of the rest of our websocket communcation to go through services like this.

File: frontend/src/events.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+// Enum of application events
```
Comment (sam-ao) [inline frontend/src/events.js:1]: I don't know about having this file in the root source folder. Maybe we can create a folder to put this kind of thing in

Comment (sam-ao) [reply frontend/src/events.js:1]: I guess the same could be said for a few other files in the root src folder

File: frontend/src/services/eventHandlers/hubHandler.js (hunk 3 above)
Comment (gordon-cheung) [reply frontend/src/services/eventHandlers/hubHandler.js:4]: Its for handling websocket events from 'hub' api

File: frontend/src/services/eventHandlers/gameHandler.js (hunk 4 above)
Comment (gordon-cheung) [reply frontend/src/services/eventHandlers/gameHandler.js:5]: Game master? I think its to handle receive events

File: frontend/src/services/gameApiService.js (hunk 7 above)
Comment (gordon-cheung) [reply frontend/src/services/gameApiService.js:1]: So like a chatApiService? or a drawApiService?

Comment (gordon-cheung) [reply frontend/src/services/gameApiService.js:1]: I havnt really investigated too much into how to actually use those services, right now they are a bunch of singletons, which I think is ok if we dont care about the state of these services.

File: frontend/src/events.js (hunk 8 above)
Comment (gordon-cheung) [reply frontend/src/events.js:1]: Ok lets see discuss this at a time, but I probably wont address it in this PR

File: frontend/src/services/eventHandlerService.js (hunk 6 above)
Comment (gordon-cheung) [reply frontend/src/services/eventHandlerService.js]: Changed it so it is only created once

File: frontend/src/services/gameApiService.js (hunk 7 above)
Comment (sam-ao) [reply frontend/src/services/gameApiService.js:1]: I just think its nicer to not have the webSocketService being used all over the place, and have a "layer" of components that the vue components use instead

File: frontend/src/services/eventHandlers/eventHandlerFactory.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import GameHandler from "./gameHandler";
+import HubHandler from "./hubHandler";
+
+export default class EventHandlerFactory {
```
Comment (sam-ao) [inline frontend/src/services/eventHandlers/eventHandlerFactory.js:4]: *minor* not really a factory anymore

File: frontend/src/services/gameApiService.js (hunk 7 above)
Comment (gordon-cheung) [reply frontend/src/services/gameApiService.js:1]: Ok i can agree with that, I think ideally we want to move logic away from components

File: frontend/src/services/eventHandlers/eventHandlerFactory.js (hunk 9 above)
Comment (gordon-cheung) [reply frontend/src/services/eventHandlers/eventHandlerFactory.js:4]: What is it :o?

**Pull request merged**